pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ClientBuilder.java

```diff
@@ -566,4 +566,20 @@ ClientBuilder authentication(String authPluginClassName, Map<String, String> aut
      * @return
      */
     ClientBuilder socks5ProxyPassword(String socks5ProxyPassword);
+
+    /**
+     * When using the client as a proxy service, client can support the original principal.
+     *
+     * @param originalPrincipal
+     * @return the client builder instance
+     */
+    ClientBuilder originalPrincipal(String originalPrincipal);
```

Comment:
If this is used internally, it should not be exposed in public APIs.

Reply:
Websocket module requires this.